src/ui/public/filter_bar/lib/__tests__/mapMatchAll.js

```diff
@@ -0,0 +1,51 @@
+
+describe('ui/filter_bar/lib', function () {
+  describe('mapMatchAll()', function () {
+    const expect = require('expect.js');
+    const ngMock = require('ngMock');
+    let resolvePromises;
+    let mapMatchAll;
+    let filter;
+
+
+    beforeEach(ngMock.module('kibana'));
+    beforeEach(ngMock.inject(function (Private, $rootScope) {
+      resolvePromises = () => $rootScope.$apply();
```

Comment:
I get what you were doing here, but I don't think the overhead is worth it. Seeing `$scope.$apply()` in the code, it's clear what that command is doing since we're already in Angular land. Seeing `resolvePromises` means you have to go look up what that code is doing. 

Calling `$apply` is also already pretty concise.